src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala

```diff
@@ -43,6 +46,71 @@ trait JsonMethodsImplicits {
       }
     } else Left(InvalidParams(s"Quantity '${input.s}' should have 0x prefix"))
   }
+
+  private def decode(s: String): Array[Byte] = {
+    val stripped = s.replaceFirst("^0x", "")
+    val normalized = if (stripped.length % 2 == 1) "0" + stripped else stripped
+    Hex.decode(normalized)
+  }
+
+  protected def extractAddress(input: String): Either[JsonRpcError, Address] =
+    Try(Address(input)).toEither.left.map(_ => InvalidAddress)
+
+  protected def extractAddress(input: JString): Either[JsonRpcError, Address] =
+    extractAddress(input.s)
+
+  protected def extractBytes(input: String): Either[JsonRpcError, ByteString] =
+    Try(ByteString(decode(input))).toEither.left.map(_ => InvalidParams())
+
+  protected def extractBytes(input: JString): Either[JsonRpcError, ByteString] =
+    extractBytes(input.s)
+
+  protected def extractQuantity(input: JValue): Either[JsonRpcError, BigInt] =
```

Comment:
We currently have a `tryExtractQuantity` function for extracting the quantity out of a `JString`, as it is not being use in this function, maybe we should remove it and replace it with this one?